packages/overlay/src/integrations/sentry/tabs/TracesTab.tsx

```diff
@@ -91,25 +230,93 @@ export default function TracesTab() {
     setAiMode(prev => !prev);
   }, []);
 
+  const context = useSpotlightContext();
+  const { allTraces, localTraces } = useSentryTraces();
+  const [showAll, setShowAll] = useState(() => !context.experiments["sentry:focus-local-events"]);
+  const visibleTraces = showAll ? allTraces : localTraces;
+  const [searchQuery, setSearchQuery] = useState("");
+  const [activeFilters, setActiveFilters] = useState<string[]>([]);
+  const { TRACE_FILTER_CONFIGS, filteredTraces } = useTraceFiltering(visibleTraces, activeFilters, searchQuery);
```

Comment:
These all should probably be combined in a single hook that simply returns `traceData`. It would have `all`, `local`, `filtered`, `searchQuery`, `activeFilters` etc. Not sure about the name `hiddenTraceCount` as it is actually `nonLocalTraceCount`. We would then move the `showAll` logic to somewhere else (potentially a global option through the URL like `?showNonLocalTraces=1`)